lib/puppet/module/task.rb

```diff
@@ -0,0 +1,85 @@
+require 'json'
+
+class Puppet::Module
+  class Task
+    class Error < Puppet::Error; end
+    class InvalidName < Error; end
+    class InvalidFile < Error; end
+    class TaskNotFound < Error; end
+
+    FORBIDDEN_EXTENSIONS = %w{.conf .md}
+
+    def self.is_task_name?(name)
+      return true if name =~ /^[a-z][a-z0-9_]*$/
+      return false
+    end
+
+    # Determine whether a file has a legal name for either a task's executable or metadata file.
+    def self.is_tasks_filename?(path)
+      name_less_extension = File.basename(path, '.*')
```

Comment:
I can replace one of the `File.basename` calls with `Puppet::FileSystem.basename`, but the other two are trickier because the `Puppet::FileSystem` version doesn't support the suffix argument. In both of those cases, the purpose is to get the base name without the extension, so I could chop it off the string if that sounds better than what we have.